client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -33,10 +33,22 @@ module.exports = React.createClass( {
 		}
 
 		window.scrollTo( 0, 0 );
-		analytics.tracks.recordEvent( 'calypso_checkout_page_view', { saved_cards: this.props.cards.get().length } );
+		if ( this.props.cart.hasLoadedFromServer ) {
+			this.trackPageView( this.props.cart );
```

Comment:
`trackPageView` is a method on `Checkout` with access to `this.props.cart`, so it might make more sense to pass no arguments and access `this.props.cart` internally (like it does with `this.props.cards`).